api/users/views.py

```diff
@@ -792,18 +792,22 @@ def post(self, request, *args, **kwargs):
         record_id = (request.data.get('id', None) or '').lower().strip()
         if not record_id:
             raise ValidationError('Must specify record "id".')
+
         claimed_user = self.get_user(check_permissions=True)  # Ensures claimability
         if claimed_user.is_disabled:
             raise ValidationError('Cannot claim disabled account.')
-        try:
-            record_referent = Guid.objects.get(_id=record_id).referent
-        except Guid.DoesNotExist:
+
+        record_referent, version = Guid.load_referent(record_id)
+        if not record_referent:
             raise NotFound('Unable to find specified record.')
```

Comment:
hmm, record object is not versioned, so it should work with `Guid.objects.get()`. Please remove this change to see what extra tests are broken.